src/CoreDAL/UserDbRepository.cs

```diff
@@ -25,43 +26,35 @@ public async Task<UserDto> Add(UserDto user)
 
         public async Task<IEnumerable<UserDto>> GetAll()
         {
-            return await _dbContext.Users.AsNoTracking().ToListAsync();
+            return await _dbContext.Users.ToListAsync();
         }
 
         public async Task<UserDto> GetById(Guid id)
         {
-            return await _dbContext.Users.AsNoTracking().FirstOrDefaultAsync(u => u.Id == id);
+            return await _dbContext.Users.SingleOrDefaultAsync(u => u.Id == id);
         }
 
         public async Task<UserDto> UpdateById(Guid id, UserDto user)
         {
-            var dbUser = await GetById(id);
-
-            if (dbUser != null)
+            var updated = await _dbContext.Users.Where(u => u.Id == id)
+                .UpdateFromQueryAsync(u => new UserDto {FirstName = user.FirstName, LastName = user.LastName, BirthDate = user.BirthDate});
+            if (updated > 0)
             {
                 user.Id = id;
-                // why it updates without id?
-                var update = _dbContext.Users.Update(user);
                 await _dbContext.SaveChangesAsync();
-                return update.Entity;
+                return user;
             }
-
             return null;
         }
 
-        public async Task<Guid?> RemoveById(Guid id)
+        public async Task<int> RemoveById(Guid id)
         {
-            var dbUser = await GetById(id);
-            // do we need to check null here?
-            if (dbUser != null)
+            var result = await _dbContext.Users.Where(u => u.Id == id).DeleteFromQueryAsync();
```

Comment:
Try to google solution without this package

Reply:
done